multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1144,62 +1255,68 @@ def evaluate(
         else:
             return results
 
-    def semantic_search(
+    def evaluate(
         self,
-        query_data: Union[pd.DataFrame, dict, list],
-        response_data: Union[pd.DataFrame, dict, list],
-        id_mappings: Dict[str, Dict],
-        return_prob: Optional[bool] = False,
-        as_pandas: Optional[bool] = None,
+        data: Optional[Union[pd.DataFrame, dict, list]] = None,
+        query_data: Optional[list] = None,
+        response_data: Optional[list] = None,
+        id_mappings: Optional[Dict[str, Dict]] = None,
+        metrics: Optional[Union[str, List[str]]] = None,
+        return_pred: Optional[bool] = False,
+        chunk_size: Optional[int] = 1024,
+        cosine: Optional[bool] = True,
+        top_k: Optional[int] = 100,
     ):
         """
-        Perform a cosine similarity search between query data and response data.
+        Evaluate model on a test dataset.
 
         Parameters
         ----------
+        data
+            A dataframe, containing the same columns as the training data.
         query_data
-            The query data.
+            A list of queries.
         response_data
-            The response data.
+            A list of response data, in which to match the queries.
         id_mappings
-            Id-to-content mappings. The contents can be text, image, etc.
-            This is used when the dataframe contains the query/response indexes instead of their contents.
-        return_prob
-            Whether to return the probability.
-        as_pandas
-            Whether to return the output as a pandas DataFrame(Series) (True) or numpy array (False).
+             Id-to-content mappings. The contents can be text, image, etc.
+             This is used when the dataframe contains the query/response indexes instead of their contents.
+        metrics
+            A list of metric names to report.
+            If None, we only return the score for the stored `_eval_metric_name`.
+        return_pred
+            Whether to return the prediction result of each row.
+        chunk_size
+            Scan the response data by chunk_size each time. Increasing the value increases the speed, but requires more memory.
+        cosine
```

Comment:
Consider to change it to `distance` and set `distance="cosine"` by default.

Reply:
Since this is for similarity instead of distance, I have changed it to similarity_type.